### PR TITLE
Bug 1212715 - [Settings] Convert call barring settings switches to use web components

### PR DESCRIPTION
**Pull request by KevinGrandon**
https://bugzilla.mozilla.org/show_bug.cgi?id=1212715

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f338f446dce2488c2f87f4884b179943643b9c40

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6c3434e16086ffdba8ed6d2738ad1d03fe26d5b5

File: apps/settings/test/unit/_call_barring.html [hunk 1]
```diff
@@ -36,17 +33,15 @@ <h2 data-l10n-id="callBarring-incoming"> Incoming calls </h2>
   </header>
   <ul id="list-callBarring-incoming">
     <li id="li-cb-baic" data-state="off">
-      <label class="pack-switch checkbox-label">
-        <input type="checkbox" data-ignore>
-        <span data-l10n-id="callBarring-baic">All</span>
-        <small id="baic-desc"></small>
+      <gaia-switch>
```
Comment (gasolin) [inline apps/settings/test/unit/_call_barring.html]: missing data-ignore attribute

File: apps/settings/test/unit/_call_barring.html [hunk 2]
```diff
@@ -9,24 +9,21 @@ <h2 data-l10n-id="callBarring-outgoing"> Outgoing calls </h2>
...
+        <label data-l10n-id="callBarring-boic">International</label>
+        <details id="boic-desc"></details>
       </label>
     </li>
     <li id="li-cb-boicExhc" data-state="off">
-      <label class="pack-switch checkbox-label">
-        <input type="checkbox" data-ignore>
-        <span data-l10n-id="callBarring-boicExhc">International except Home Country</span>
-        <small id="boicExhc-desc"></small>
+      <gaia-switch>
```
Comment (gasolin) [inline apps/settings/test/unit/_call_barring.html]: missing data-ignore attribute

File: apps/settings/test/unit/_call_barring.html [hunk 3]
```diff
@@ -9,24 +9,21 @@ <h2 data-l10n-id="callBarring-outgoing"> Outgoing calls </h2>
...
+        <label data-l10n-id="callBarring-baoc">All</label>
+        <details id="baoc-desc"></details>
       </label>
     </li>
     <li id="li-cb-boic" data-state="off">
-      <label class="pack-switch checkbox-label">
-        <input type="checkbox" data-ignore>
-        <span data-l10n-id="callBarring-boic">International</span>
-        <small id="boic-desc"></small>
+      <gaia-switch>
```
Comment (gasolin) [inline apps/settings/test/unit/_call_barring.html]: missing data-ignore attribute

File: apps/settings/test/unit/_call_barring.html [hunk 4]
```diff
@@ -9,24 +9,21 @@ <h2 data-l10n-id="callBarring-outgoing"> Outgoing calls </h2>
   </header>
   <ul id="list-callBarring-outgoing">
     <li id="li-cb-baoc" data-state="off">
-      <label class="pack-switch checkbox-label">
-        <input type="checkbox" data-ignore>
-        <span data-l10n-id="callBarring-baoc">All</span>
-        <small id="baoc-desc"></small>
+      <gaia-switch>
```
Comment (gasolin) [inline apps/settings/test/unit/_call_barring.html]: missing data-ignore attribute

File: apps/settings/test/unit/_call_barring.html [hunk 5]
```diff
@@ -36,17 +33,15 @@ <h2 data-l10n-id="callBarring-incoming"> Incoming calls </h2>
...
+        <label data-l10n-id="callBarring-baic">All</label>
+        <details id="baic-desc"></details>
       </label>
     </li>
     <li id="li-cb-baicR" data-state="off">
-      <label class="pack-switch checkbox-label">
-        <input type="checkbox" data-ignore>
-        <span data-l10n-id="callBarring-baicR">All while roaming</span>
-        <small id="bairR-desc"></small>
+      <gaia-switch>
```
Comment (gasolin) [inline apps/settings/test/unit/_call_barring.html]: missing data-ignore attribute

File: apps/settings/test/unit/_call_barring.html (hunk 4 above)
Comment (gasolin) [inline apps/settings/test/unit/_call_barring.html]: ooh its a unit test file, you can ignore that attribute...

**Pull request closed without merging**